pkg/vault/test/authorizer/TimelockAuthorizerExecute.test.ts

```diff
@@ -370,6 +370,17 @@ describe('TimelockAuthorizer execute', () => {
       await expect(authorizer.execute(id, { from: executor })).to.be.revertedWith('ACTION_ALREADY_EXECUTED');
     });
 
+    it('cannot execute another scheculed action within a scheduled action', async () => {
+      const id = await schedule();
+
+      const data = authorizer.interface.encodeFunctionData('execute', [id]);
+      const nextId = await authorizer.schedule(authorizer, data, [], { from: user });
+
+      await advanceTime(delay);
+
+      await expect(authorizer.execute(nextId, { from: executor })).to.be.revertedWith('CANNOT_SCHEDULE_AUTHORIZER_ACTIONS');
```

Comment:
Note that this revert is not what you're trying to test: you're attempting to schedule an authorizer call directly (which is not allowed). We actually already have a test for this.

But I think we should maybe remove the reentrancy guard check altogether.